src/Orleans.Runtime/Silo/Watchdog.cs

```diff
@@ -32,36 +33,44 @@ public void Start()
         {
             logger.LogInformation("Starting Silo Watchdog.");
 
-            if (thread is not null)
+            if (selfThread is not null)
             {
                 throw new InvalidOperationException("Watchdog.Start may not be called more than once");
             }
 
             var now = DateTime.UtcNow;
             lastHeartbeat = now;
-            lastWatchdogCheck = now;
             cumulativeGCPauseDuration = GC.GetTotalPauseDuration();
 
-            this.thread = new Thread(this.Run)
+            this.selfThread = new Thread(this.RunSelfCheck)
+            {
+                IsBackground = true,
+                Name = "Orleans.Runtime.Watchdog",
+            };
+            this.selfThread.Start();
+
+            lastWatchdogCheck = now;
+
+            this.participantThread = new Thread(this.RunParticipantCheck)
             {
                 IsBackground = true,
                 Name = "Orleans.Runtime.Watchdog",
```

Comment:
These two threads have the same name but different behavior.

Reply:
Changed that, thanks.